### PR TITLE
Using spark in local mode

### PR DESCRIPTION
**Pull request by smadha**
@chrismattmann - Can you see if such thing is fine for Tika? 

This needs some work in terms of returning age etc.. but I wanted to share the idea.

I just pulled out lines of code from [AgePredictTool.java](https://github.com/smadha/AgePredictor/blob/fa2a5f23861a988858b3f10166f40b29b7a7595f/src/main/java/gov/nasa/jpl/ml/cmdline/spark/authorage/AgePredictTool.java) and made it work such that it takes a [string](https://github.com/smadha/AgePredictor/blob/fa2a5f23861a988858b3f10166f40b29b7a7595f/src/main/java/gov/nasa/jpl/ml/cmdline/spark/authorage/AgePredicterLocal.java#L64) and print a [number](https://github.com/smadha/AgePredictor/blob/fa2a5f23861a988858b3f10166f40b29b7a7595f/src/main/java/gov/nasa/jpl/ml/cmdline/spark/authorage/AgePredicterLocal.java#L111)

### CODE REVIEW

**Comment by smadha** [comment]
@chrismattmann Try it now

```bash
cd AgePredictor
mvn clean install
java -cp age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar edu.usc.irds.agepredictor.authorage.AgePredicterLocal I am actually very young now
```

**Comment by chrismattmann** [comment]
Hey @smadha OK built fine:

```
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ age-predictor-assembly ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:3.0.0:jar (default-jar) @ age-predictor-assembly ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: /Users/mattmann/git/AgePredictor/age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-assembly-plugin:2.4:single (make-assembly) @ age-predictor-assembly ---
[INFO] Building jar: /Users/mattmann/git/AgePredictor/age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ age-predictor-assembly ---
[INFO] Installing /Users/mattmann/git/AgePredictor/age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT.jar to /Users/mattmann/.m2/repository/gov/nasa/jpl/ml/age-predictor-assembly/1.0-SNAPSHOT/age-predictor-assembly-1.0-SNAPSHOT.jar
[INFO] Installing /Users/mattmann/git/AgePredictor/age-predictor-assembly/pom.xml to /Users/mattmann/.m2/repository/gov/nasa/jpl/ml/age-predictor-assembly/1.0-SNAPSHOT/age-predictor-assembly-1.0-SNAPSHOT.pom
[INFO] Installing /Users/mattmann/git/AgePredictor/age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar to /Users/mattmann/.m2/repository/gov/nasa/jpl/ml/age-predictor-assembly/1.0-SNAPSHOT/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] age-classify ....................................... SUCCESS [  0.866 s]
[INFO] age-predictor-opennlp .............................. SUCCESS [  2.011 s]
[INFO] age-predictor-cli .................................. SUCCESS [  7.507 s]
[INFO] age-predictor-api .................................. SUCCESS [  4.326 s]
[INFO] age-predictor-assembly ............................. SUCCESS [ 27.560 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 42.373 s
[INFO] Finished at: 2017-05-10T11:40:35-07:00
[INFO] Final Memory: 139M/1479M
[INFO] ------------------------------------------------------------------------
LMC-053601:AgePredictor mattmann$ 
```

Running your tests didn't work though?

```
LMC-053601:AgePredictor mattmann$ java -cp age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar edu.usc.irds.agepredictor.authorage.AgePredicterLocal I am actually very young now
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
17/05/10 11:41:09 INFO SparkContext: Running Spark version 2.0.0
17/05/10 11:41:09 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/05/10 11:41:09 INFO SecurityManager: Changing view acls to: mattmann
17/05/10 11:41:09 INFO SecurityManager: Changing modify acls to: mattmann
17/05/10 11:41:09 INFO SecurityManager: Changing view acls groups to: 
17/05/10 11:41:09 INFO SecurityManager: Changing modify acls groups to: 
17/05/10 11:41:09 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(mattmann); groups with view permissions: Set(); users  with modify permissions: Set(mattmann); groups with modify permissions: Set()
17/05/10 11:41:10 INFO Utils: Successfully started service 'sparkDriver' on port 65192.
17/05/10 11:41:10 INFO SparkEnv: Registering MapOutputTracker
17/05/10 11:41:10 INFO SparkEnv: Registering BlockManagerMaster
17/05/10 11:41:10 INFO DiskBlockManager: Created local directory at /private/var/folders/n5/1d_k3z4s2293q8ntx_n8sw54mm5n_8/T/blockmgr-a75added-3589-4d17-9549-0da219de8b39
17/05/10 11:41:10 INFO MemoryStore: MemoryStore started with capacity 2004.6 MB
17/05/10 11:41:10 INFO SparkEnv: Registering OutputCommitCoordinator
17/05/10 11:41:10 INFO Utils: Successfully started service 'SparkUI' on port 4040.
17/05/10 11:41:10 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://192.168.1.65:4040
17/05/10 11:41:10 INFO Executor: Starting executor ID driver on host localhost
17/05/10 11:41:10 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 65194.
17/05/10 11:41:10 INFO NettyBlockTransferService: Server created on 192.168.1.65:65194
17/05/10 11:41:10 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 192.168.1.65, 65194)
17/05/10 11:41:10 INFO BlockManagerMasterEndpoint: Registering block manager 192.168.1.65:65194 with 2004.6 MB RAM, BlockManagerId(driver, 192.168.1.65, 65194)
17/05/10 11:41:10 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 192.168.1.65, 65194)
17/05/10 11:41:10 WARN SparkContext: Use an existing SparkContext, some configuration may not take effect.
17/05/10 11:41:10 INFO SharedState: Warehouse path is 'file:/Users/mattmann/git/AgePredictor/spark-warehouse'.
Error with loading Sentence model...
java.io.FileNotFoundException: model/opennlp/en-sent.bin (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at opennlp.tools.tokenize.SentenceTokenizer.tokenize(SentenceTokenizer.java:49)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.predictAge(AgePredicterLocal.java:68)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.main(AgePredicterLocal.java:127)
java.io.FileNotFoundException: model/opennlp/en-pos-maxent.bin (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at opennlp.tools.util.featuregen.POSFeatureGenerator.extractFeatures(POSFeatureGenerator.java:48)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.predictAge(AgePredicterLocal.java:76)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.main(AgePredicterLocal.java:127)
17/05/10 11:41:17 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 6.1 MB, free 1998.5 MB)
17/05/10 11:41:18 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 488.5 KB, free 1998.0 MB)
17/05/10 11:41:18 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 192.168.1.65:65194 (size: 488.5 KB, free: 2004.1 MB)
17/05/10 11:41:18 INFO SparkContext: Created broadcast 0 from broadcast at CountVectorizer.scala:243
17/05/10 11:41:20 INFO CodeGenerator: Code generated in 110.086679 ms
17/05/10 11:41:20 INFO SparkContext: Starting job: first at AgePredicterLocal.java:105
17/05/10 11:41:20 INFO DAGScheduler: Got job 0 (first at AgePredicterLocal.java:105) with 1 output partitions
17/05/10 11:41:20 INFO DAGScheduler: Final stage: ResultStage 0 (first at AgePredicterLocal.java:105)
17/05/10 11:41:20 INFO DAGScheduler: Parents of final stage: List()
17/05/10 11:41:20 INFO DAGScheduler: Missing parents: List()
17/05/10 11:41:20 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at javaRDD at AgePredicterLocal.java:103), which has no missing parents
17/05/10 11:41:20 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 7.7 KB, free 1998.0 MB)
17/05/10 11:41:20 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 4.1 KB, free 1998.0 MB)
17/05/10 11:41:20 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 192.168.1.65:65194 (size: 4.1 KB, free: 2004.1 MB)
17/05/10 11:41:20 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1012
17/05/10 11:41:20 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at javaRDD at AgePredicterLocal.java:103)
17/05/10 11:41:20 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks
17/05/10 11:41:20 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0, localhost, partition 0, PROCESS_LOCAL, 5352 bytes)
17/05/10 11:41:20 INFO Executor: Running task 0.0 in stage 0.0 (TID 0)
17/05/10 11:41:20 INFO CodeGenerator: Code generated in 15.124857 ms
17/05/10 11:41:20 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 976 bytes result sent to driver
17/05/10 11:41:20 INFO TaskSetManager: Finished task 0.0 in stage 0.0 (TID 0) in 80 ms on localhost (1/1)
17/05/10 11:41:20 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool 
17/05/10 11:41:20 INFO DAGScheduler: ResultStage 0 (first at AgePredicterLocal.java:105) finished in 0.091 s
17/05/10 11:41:20 INFO DAGScheduler: Job 0 finished: first at AgePredicterLocal.java:105, took 0.149709 s
Exception in thread "main" java.lang.UnsupportedOperationException: empty collection
	at org.apache.spark.rdd.RDD$$anonfun$first$1.apply(RDD.scala:1321)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:358)
	at org.apache.spark.rdd.RDD.first(RDD.scala:1318)
	at org.apache.spark.api.java.JavaRDDLike$class.first(JavaRDDLike.scala:537)
	at org.apache.spark.api.java.AbstractJavaRDDLike.first(JavaRDDLike.scala:45)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.predictAge(AgePredicterLocal.java:105)
	at edu.usc.irds.agepredictor.authorage.AgePredicterLocal.main(AgePredicterLocal.java:127)
17/05/10 11:41:20 INFO SparkContext: Invoking stop() from shutdown hook
17/05/10 11:41:20 INFO SparkUI: Stopped Spark web UI at http://192.168.1.65:4040
17/05/10 11:41:20 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
17/05/10 11:41:20 INFO MemoryStore: MemoryStore cleared
17/05/10 11:41:20 INFO BlockManager: BlockManager stopped
17/05/10 11:41:20 INFO BlockManagerMaster: BlockManagerMaster stopped
17/05/10 11:41:20 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
17/05/10 11:41:20 INFO SparkContext: Successfully stopped SparkContext
17/05/10 11:41:20 INFO ShutdownHookManager: Shutdown hook called
17/05/10 11:41:20 INFO ShutdownHookManager: Deleting directory /private/var/folders/n5/1d_k3z4s2293q8ntx_n8sw54mm5n_8/T/spark-7c0c116a-12e5-4250-b92a-90c3bd635f53
LMC-053601:AgePredictor mattmann$ 

```

ideas?

**Comment by smadha** [comment]
Ah, you need to download `en-pos-maxent.bin`, `en-sent.bin` and `en-token.bin` from http://opennlp.sourceforge.net/models-1.5/ to `model/opennlp/`

https://github.com/USCDataScience/AgePredictor#downloads

Do you think committing opennlp model in this repo?

**Comment by chrismattmann** [comment]
Steal Thamme code from tika that downloads automatically on build

**Comment by smadha** [comment]
I did it little differently through a maven plugin to download files. Should work automatically by executing `mvn clean install`

**Comment by chrismattmann** [comment]
awesome, great work. So, it builds, and works! 

```
Downloading: http://opennlp.sourceforge.net/models-1.5/en-pos-maxent.bin
5562K downloaded
[INFO] 
[INFO] --- download-maven-plugin:1.3.0:wget (download-en-sent) @ age-predictor-assembly ---
Downloading: http://opennlp.sourceforge.net/models-1.5/en-sent.bin
96K downloaded
[INFO] 
[INFO] --- download-maven-plugin:1.3.0:wget (download-en-token) @ age-predictor-assembly ---
Downloading: http://opennlp.sourceforge.net/models-1.5/en-token.bin
429K downloaded
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] age-predictor ...................................... SUCCESS [  1.060 s]
[INFO] age-predictor-opennlp .............................. SUCCESS [  2.246 s]
[INFO] age-predictor-cli .................................. SUCCESS [  8.442 s]
[INFO] age-predictor-api .................................. SUCCESS [  4.773 s]
[INFO] age-predictor-assembly ............................. SUCCESS [ 49.926 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:06 min
[INFO] Finished at: 2017-05-10T21:10:15-07:00
[INFO] Final Memory: 66M/1514M
[INFO] ------------------------------------------------------------------------
LMC-053601:AgePredictor mattmann$ !java
java -cp age-predictor-assembly/target/age-predictor-assembly-1.0-SNAPSHOT-jar-with-dependencies.jar edu.usc.irds.agepredictor.authorage.AgePredicterLocal I am actually very young now
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
17/05/10 21:10:20 INFO SparkContext: Running Spark version 2.0.0
17/05/10 21:10:20 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/05/10 21:10:20 INFO SecurityManager: Changing view acls to: mattmann
17/05/10 21:10:20 INFO SecurityManager: Changing modify acls to: mattmann
17/05/10 21:10:20 INFO SecurityManager: Changing view acls groups to: 
17/05/10 21:10:20 INFO SecurityManager: Changing modify acls groups to: 
17/05/10 21:10:20 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(mattmann); groups with view permissions: Set(); users  with modify permissions: Set(mattmann); groups with modify permissions: Set()
17/05/10 21:10:20 INFO Utils: Successfully started service 'sparkDriver' on port 53538.
17/05/10 21:10:20 INFO SparkEnv: Registering MapOutputTracker
17/05/10 21:10:20 INFO SparkEnv: Registering BlockManagerMaster
17/05/10 21:10:20 INFO DiskBlockManager: Created local directory at /private/var/folders/n5/1d_k3z4s2293q8ntx_n8sw54mm5n_8/T/blockmgr-f46b8dd4-0b77-4522-8daf-4212a651aa2d
17/05/10 21:10:20 INFO MemoryStore: MemoryStore started with capacity 2004.6 MB
17/05/10 21:10:20 INFO SparkEnv: Registering OutputCommitCoordinator
17/05/10 21:10:20 INFO Utils: Successfully started service 'SparkUI' on port 4040.
17/05/10 21:10:20 INFO SparkUI: Bound SparkUI to 0.0.0.0, and started at http://172.20.7.63:4040
17/05/10 21:10:21 INFO Executor: Starting executor ID driver on host localhost
17/05/10 21:10:21 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 53539.
17/05/10 21:10:21 INFO NettyBlockTransferService: Server created on 172.20.7.63:53539
17/05/10 21:10:21 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 172.20.7.63, 53539)
17/05/10 21:10:21 INFO BlockManagerMasterEndpoint: Registering block manager 172.20.7.63:53539 with 2004.6 MB RAM, BlockManagerId(driver, 172.20.7.63, 53539)
17/05/10 21:10:21 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 172.20.7.63, 53539)
17/05/10 21:10:21 WARN SparkContext: Use an existing SparkContext, some configuration may not take effect.
17/05/10 21:10:21 INFO SharedState: Warehouse path is 'file:/Users/mattmann/git/AgePredictor/spark-warehouse'.
17/05/10 21:10:29 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 6.1 MB, free 1998.5 MB)
17/05/10 21:10:30 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 488.5 KB, free 1998.0 MB)
17/05/10 21:10:30 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.20.7.63:53539 (size: 488.5 KB, free: 2004.1 MB)
17/05/10 21:10:30 INFO SparkContext: Created broadcast 0 from broadcast at CountVectorizer.scala:243
17/05/10 21:10:32 INFO CodeGenerator: Code generated in 102.449019 ms
17/05/10 21:10:32 INFO SparkContext: Starting job: first at AgePredicterLocal.java:105
17/05/10 21:10:32 INFO DAGScheduler: Got job 0 (first at AgePredicterLocal.java:105) with 1 output partitions
17/05/10 21:10:32 INFO DAGScheduler: Final stage: ResultStage 0 (first at AgePredicterLocal.java:105)
17/05/10 21:10:32 INFO DAGScheduler: Parents of final stage: List()
17/05/10 21:10:32 INFO DAGScheduler: Missing parents: List()
17/05/10 21:10:32 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at javaRDD at AgePredicterLocal.java:103), which has no missing parents
17/05/10 21:10:32 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 10.9 KB, free 1998.0 MB)
17/05/10 21:10:32 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 5.4 KB, free 1998.0 MB)
17/05/10 21:10:32 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 172.20.7.63:53539 (size: 5.4 KB, free: 2004.1 MB)
17/05/10 21:10:32 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1012
17/05/10 21:10:32 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at javaRDD at AgePredicterLocal.java:103)
17/05/10 21:10:32 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks
17/05/10 21:10:32 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0, localhost, partition 0, PROCESS_LOCAL, 6671 bytes)
17/05/10 21:10:32 INFO Executor: Running task 0.0 in stage 0.0 (TID 0)
17/05/10 21:10:32 INFO CodeGenerator: Code generated in 23.093846 ms
17/05/10 21:10:32 INFO Executor: Finished task 0.0 in stage 0.0 (TID 0). 3381 bytes result sent to driver
17/05/10 21:10:32 INFO TaskSetManager: Finished task 0.0 in stage 0.0 (TID 0) in 80 ms on localhost (1/1)
17/05/10 21:10:32 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool 
17/05/10 21:10:32 INFO DAGScheduler: ResultStage 0 (first at AgePredicterLocal.java:105) finished in 0.094 s
17/05/10 21:10:32 INFO DAGScheduler: Job 0 finished: first at AgePredicterLocal.java:105, took 0.141711 s

===================

Text received- 'I am actually very young now ' 
 Predicted Age - 34.983567


===================

17/05/10 21:10:32 INFO SparkContext: Invoking stop() from shutdown hook
17/05/10 21:10:32 INFO SparkUI: Stopped Spark web UI at http://172.20.7.63:4040
17/05/10 21:10:32 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
17/05/10 21:10:32 INFO MemoryStore: MemoryStore cleared
17/05/10 21:10:32 INFO BlockManager: BlockManager stopped
17/05/10 21:10:32 INFO BlockManagerMaster: BlockManagerMaster stopped
17/05/10 21:10:32 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
17/05/10 21:10:32 INFO SparkContext: Successfully stopped SparkContext
17/05/10 21:10:32 INFO ShutdownHookManager: Shutdown hook called
17/05/10 21:10:32 INFO ShutdownHookManager: Deleting directory /private/var/folders/n5/1d_k3z4s2293q8ntx_n8sw54mm5n_8/T/spark-7900868d-3141-4c13-b851-c97b044f0e63
LMC-053601:AgePredictor mattmann$ 
```
Merging now great work.

**Comment by smadha** [comment]
coming soon Tika AgeParser..

**Comment by chrismattmann** [comment]
lovely @smadha

**Comment by chrismattmann** [comment]
make it an AgeRecogniser...my recommendation 👍

File: age-predictor-assembly/pom.xml [hunk 1]
```diff
@@ -97,6 +101,46 @@
...
+						<goals>
+							<goal>wget</goal>
+						</goals>
+						<configuration>
+							<url>http://opennlp.sourceforge.net/models-1.5/en-token.bin</url>
+							<outputDirectory>${model.opennlp}</outputDirectory>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
```
Comment (smadha) [inline age-predictor-assembly/pom.xml]: @thammegowda - Did you see this? Maybe an alternative approach for downloading models in Tika?

Comment (thammegowda) [inline age-predictor-assembly/pom.xml]: I did. :+1:  Some people like XML some like scripts 😉  Best to have examples for both

**Comment by chrismattmann** [comment]
FWIW, OpenNLP models are only needed for training:

```
age-predictor-opennlp/src/main/java/opennlp/tools/tokenize/SentenceTokenizer.java:    private final String TOKEN_MODEL_FILE = "model/opennlp/en-token.bin";
age-predictor-opennlp/src/main/java/opennlp/tools/tokenize/SentenceTokenizer.java:    private final String SENTENCE_MODEL_FILE = "model/opennlp/en-sent.bin";
age-predictor-opennlp/src/main/java/opennlp/tools/util/featuregen/POSFeatureGenerator.java:    private final String MODEL_FILE = "model/opennlp/en-pos-maxent.bin";
```

**Comment by smadha** [comment]
SentenceTokenizer is needed while predicting to tokenize     
  
Model used- 
https://github.com/USCDataScience/AgePredictor/blob/master/age-predictor-opennlp/src/main/java/opennlp/tools/tokenize/SentenceTokenizer.java#L49

Code calling Tokenizer - 
https://github.com/USCDataScience/AgePredictor/blob/master/age-predictor-api/src/main/java/edu/usc/irds/agepredictor/authorage/AgePredicterLocal.java#L77

**Pull request merged**